src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.BadRequests.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+
+internal sealed partial class KestrelTrace : ILogger
+{
+    public void ConnectionBadRequest(string connectionId, AspNetCore.Http.BadHttpRequestException ex)
+    {
+        BadRequestsLog.ConnectionBadRequest(_badRequestsLogger, connectionId, ex.Message, ex);
+    }
+
+    public void RequestProcessingError(string connectionId, Exception ex)
+    {
+        BadRequestsLog.RequestProcessingError(_badRequestsLogger, connectionId, ex);
+    }
+
+    public void RequestBodyMinimumDataRateNotSatisfied(string connectionId, string? traceIdentifier, double rate)
+    {
+        BadRequestsLog.RequestBodyMinimumDataRateNotSatisfied(_badRequestsLogger, connectionId, traceIdentifier, rate);
+    }
+
+    public void ResponseMinimumDataRateNotSatisfied(string connectionId, string? traceIdentifier)
+    {
+        BadRequestsLog.ResponseMinimumDataRateNotSatisfied(_badRequestsLogger, connectionId, traceIdentifier);
+    }
+
+    public void PossibleInvalidHttpVersionDetected(string connectionId, HttpVersion expectedHttpVersion, HttpVersion detectedHttpVersion)
+    {
+        if (_generalLogger.IsEnabled(LogLevel.Debug))
+        {
+            BadRequestsLog.PossibleInvalidHttpVersionDetected(_badRequestsLogger, connectionId, HttpUtilities.VersionToString(expectedHttpVersion), HttpUtilities.VersionToString(detectedHttpVersion));
+        }
+    }
+
+    public static partial class BadRequestsLog
```

Comment:
Thanks, I went ahead and fixed this. I thought at first this wouldn't compile, but that was because I had set the methods inside to private as well as the class.